src/games/strategy/triplea/ai/fastAI/FastAI.java

```diff
@@ -9,14 +9,16 @@
  */
 public class FastAI extends ProAI {
 
+  // Odds estimator
+  private final static IOddsCalculator estimator = new FastOddsEstimator();
+
   public FastAI(final String name, final String type) {
     super(name, type);
   }
 
   @Override
-  public void initializeBattleCalculator() {
-    final IOddsCalculator calc = new FastOddsEstimator();
-    ProBattleUtils.setOddsCalculator(calc);
+  protected void initializeCalc() {
+    calc = new ProOddsCalculator(estimator);
```

Comment:
Few code smells:
- tight coupling between this class and the calculator implementation
- class initialization done outside of the constructor. 
- programming via side effects, the call to 'initializeCalc' has side effects
- unclear control flow, favoring inheritance when composition is more appropriate.

For example, looking at the ProAI code, and the FastAI calc, it's not at all obvious the at the super call will end up returning back to the 'initializeCalc', which overrides the ProAI. Looking at the ProAI, there is coupling in the constructor code and what can be overriden. This is a potential violation of the Liskov substitution principle.

I think this is very simply solved by making "IOddsCalculator" be a constructor argument. You would be following the dependency injection pattern there, and that's a good thing. It will also allow you to decouple the testing logic, if you test FastAI or ProAI, you also have to black box via side effects test the calculator too, which is not trivial nor clean (not even a unit test then as well)